experimental/dds/tree2/src/class-tree/schemaTypes.ts

```diff
@@ -0,0 +1,283 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { MakeNominal, RestrictiveReadonlyRecord } from "../util";
+import { FlexListToUnion, LazyItem } from "../feature-libraries";
+import { TreeListNodeBase, Unhydrated } from "../simple-tree";
+
+/**
+ * Base type which all nodes extend.
+ * @alpha
+ */
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+export class NodeBase {}
+
+/**
+ * A {@link TreeNode} which implements 'readonly T[]' and the list mutation APIs.
+ * @alpha
+ */
+export interface TreeListNode<TTypes extends ImplicitAllowedTypes = ImplicitAllowedTypes>
+	extends TreeListNodeBase<
+		TreeNodeFromImplicitAllowedTypes<TTypes>,
+		Unhydrated<TreeNodeFromImplicitAllowedTypes<TTypes>>, // TODO: insertion type.
+		TreeListNode
+	> {}
+
+/**
+ * Helper used to produce types for object nodes.
+ * @alpha
+ */
+export type ObjectFromSchemaRecord<
+	T extends RestrictiveReadonlyRecord<string, ImplicitFieldSchema>,
+> = {
+	-readonly [Property in keyof T]: TreeFieldFromImplicitField<T[Property]>;
+};
+
+/**
+ * Helper used to produce types for:
```

Comment:
Nit: to make this render as a markdown list, you'll need blank lines between each of the entries (and before the first one). TSDoc is spec'd to treat single line breaks as spaces.
https://github.com/microsoft/FluidFramework/wiki/TSDoc-Guidelines#line-breaks

Reply:
Ah, so ts-doc only supports markdown loose lists. Intelisense supports tight lists just fine, but I'll adjust accordingly.

Reply:
Done.